src/main/scala/firrtl/ExecutionOptionsManager.scala

```diff
@@ -0,0 +1,327 @@
+  }
+}
+
+trait HasFirrtlOptions {
+  self: ExecutionOptionsManager =>
+  var firrtlOptions = FirrtlExecutionOptions()
+
+  parser.note("firrtl options")
+
+  parser.opt[String]("input-file")
+    .abbr("fif")
+    .valueName ("<firrtl-source>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(inputFileNameOverride = x)
+    }.text {
+      "use this to override the top name default"
+    }
+
+  parser.opt[String]("output-file")
+    .abbr("fof")
+    .valueName ("<output>").
+    foreach { x =>
+      firrtlOptions = firrtlOptions.copy(outputFileNameOverride = x)
+    }.text {
+      "use this to override the default name"
+    }
+
+  parser.opt[String]("compiler")
+    .abbr("fc")
+    .valueName ("<high|low|verilog>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(compilerName = x)
+    }
+    .validate { x =>
+      if (Array("high", "low", "verilog").contains(x.toLowerCase)) parser.success
+      else parser.failure(s"$x not a legal compiler")
+    }.text {
+      "compiler to use, default is verilog"
+    }
+
+  parser.opt[String]("info-mode")
+    .valueName ("<ignore|use|gen|append>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(infoModeName = x.toLowerCase)
+    }
+    .validate { x =>
+      if (Array("ignore", "use", "gen", "append").contains(x.toLowerCase)) parser.success
+      else parser.failure(s"$x not a legal compiler")
+    }
+    .text {
+      "specifies the source info handling"
+    }
+
+  parser.opt[Seq[String]]("in-line")
+    .abbr("fil")
+    .valueName ("<circuit>[.<module>[.<instance>]][,..],")
+    .foreach { x =>
+      val newAnnotations = x.map { value =>
+        value.split('.') match {
+          case Array(circuit) =>
+            passes.InlineAnnotation(CircuitName(circuit), TransID(0))
+          case Array(circuit, module) =>
+            passes.InlineAnnotation(ModuleName(module, CircuitName(circuit)), TransID(0))
+          case Array(circuit, module, inst) =>
+            passes.InlineAnnotation(ComponentName(inst, ModuleName(module, CircuitName(circuit))), TransID(0))
+        }
+      }
+      firrtlOptions = firrtlOptions.copy(annotations = firrtlOptions.annotations ++ newAnnotations)
+    }
+    .text {
+      """Inline one or more module (comma separated, no spaces) module looks like "MyModule" or "MyModule.myinstance"""
+    }
+
+  parser.opt[Seq[String]]("infer-rw")
+    .abbr("firw")
+    .valueName ("<circuit[,...]>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(
+        annotations = firrtlOptions.annotations ++
+          x.map { value => passes.InferReadWriteAnnotation(value, TransID(-1)) }
+      )
+    }.text {
+      "Enable readwrite port inference for the target circuit, for multiples separate with commas and no spaces"
+    }
+
+  parser.opt[Seq[String]]("repl-seq-mem")
+    .abbr("frsq")
+    .valueName ("-c:<circuit>:-i:<filename>:-o:<filename>[,...]")
```

Comment:
I don't understand comment.  If this option can have multiple values then scopt wants them to be comma separated.  Or are you saying there cannot be multiple values